src/actions/roles.js

```diff
@@ -60,7 +60,12 @@ export function setPendingRoles(roles) {
 export function loadRolesIfNeeded(roles) {
     return async (dispatch, getState) => {
         const state = getState();
-        const pendingRoles = new Set(state.entities.roles.pending);
+        let pendingRoles = new Set();
+        try {
+            pendingRoles = new Set(state.entities.roles.pending);
+        } catch (e) {
+            // eslint-disable-line
```

Comment:
Can we include a `console.warn` here? I don't think we should completely silence this error

Reply:
I'm ok with that as long as someone actually investigates why sometimes the value of `...roles.pending` is an object

Reply:
We know it's happening a lot; we know we need to investigate. What's the value of adding a warning to everyone's console?

Reply:
I'm 0/5 on the decision

Reply:
I'd prefer to have a warning so that we actually go and fix the root problem here. Ideally, we'd fix whatever is causing `state.entities.roles.pending` to be an object right now, but I understand there's not time to do that for 1.9

Reply:
Does it maybe need to be added to the set transformer in `app/store/index.js` like the transformers for the other Set parts of the reducer?

Reply:
@hmhealey that is another PR that goes along with this one that I have ready to submit once this one gets merged